### PR TITLE
test(linter/unicorn): add fixer tests for `no-null`

### PR DESCRIPTION
**Pull request by DonIsaac**

### CODE REVIEW

**Comment by DonIsaac** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5173?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5175** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5175?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5174** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5174?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5173** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5173?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5172** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5172?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/unicorn/no_null.rs [hunk 1]
```diff
@@ -303,5 +305,45 @@ fn test() {
...
-    Tester::new(NoNull::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("let x = null;", "let x;", None),
+        ("let x = null as any;", "let x = undefined as any;", None),
+        ("if (foo == null) {}", "if (foo == undefined) {}", None),
+        ("if (foo != null) {}", "if (foo != undefined) {}", None),
+        ("if (foo == null) {}", "if (foo == undefined) {}", Some(check_strict_equality(true))),
+        // FIXME
+        (
+            "if (foo === null || foo === undefined) {}",
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/no_null.rs:316]: i wonder whether when the `null` literal is in a comparison (while condition/if condition/switch condition/binary expression (equality)) we just shouldn't fix it - it's probably gonna break people's code

Comment (DonIsaac) [reply crates/oxc_linter/src/rules/unicorn/no_null.rs:316]: I'm thinking something similar. That, or we make it a suggestion.

**Review by camc314 — APPROVED** [review]
